daemon/cdi_other.go

```diff
@@ -0,0 +1,17 @@
+//go:build !linux
+// +build !linux
+
+package daemon
+
+import "github.com/container-orchestrated-devices/container-device-interface/pkg/cdi"
+
+// RegisterCDIDriver is a no-op on non-Linux platforms.
+//
+// Note that CDI is not inherrently linux-specific, there are some linux-specific assumptions / implementations in the code that
+// queries the properties of device on the host as wel as performs the injection of device nodes and their access permissions into the OCI spec.
+//
+// In order to lift this restruction the following would have to be addressed:
+// - Support needs to be added to the cdi package for injecting Windows devices: https://github.com/container-orchestrated-devices/container-device-interface/issues/28
+// - The DeviceRequests API must be extended to non-linux platforms.
```

Comment:
While these are good reasons not to register the CDI driver on platforms other than Linux, they are not technical limitations which prevent the real `RegisterCDIDriver` function from being _successfully compiled_ on other platforms. As far as I can see, the only technical limitation is that the file `daemon/devices_linux.go` is not named `daemon/devices.go`. That limitation is entirely artificial: there isn't anything in that file which is platform-specific.

Reply:
To be clear, are you suggesting renaming `daemon/devices_linux.go` to `daemon/devices.go` in this PR? Or only changing the comment and punting that change towards a wider refactoring using pluginStore (and pluginManager?) ala https://github.com/moby/moby/pull/45134#discussion_r1142384094?

Reply:
I am happy to rename the `devices_linux.go` file in the PR if we want to see that as the first step in the refactoring.